src/application/clients/crux-service-client.ts

```diff
@@ -0,0 +1,28 @@
+import { CruxProtocolMessenger, RemoteKeyManager} from "../../core/domain-services";
+import { keyManagementProtocol } from "../../infrastructure";
+import { CruxId, InMemStorage } from "../../packages/";
+import { CruxWalletClient } from "./crux-wallet-client";
+
+export class CruxServiceClient {
+    private selfIdClaim: any;
+    private cruxProtocolMessenger: any;
+
+    constructor(selfIdClaim: any, secureCruxNetwork: any, protocolSchema?: any) {
```

Comment:
Doesn't make sense to take protocolSchema as optional parameter. You always need protocol schema to be Key Management Protocol. 
Is there any use case you designed this for? If not, remove.

Reply:
No, it does make sense I guess  because CruxServiceClient can give you the different type of clients and some might use protocolSchema some might not

Reply:
Lets add this when we add more use cases to CruxServiceClient. for now remove.

Reply:
done

Reply:
Please remove `protocolSchema` from the constructor itself. Making it non optional doesn't help us. The point is that there is only 1 protocolSchema this class can be used with at the moment. does not make sense for caller to have to send it.

Just like in CruxWalletClient, the caller does NOT have any awareness of protocol schema or messenger internals or anything.

Reply:
Okay i will remove it